### DISCUSSION TITLE
ScriptReference/EditorWindow

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#502] [Notes]
# ScriptReference/EditorWindow

https://docs.unity3d.com/ScriptReference/EditorWindow.html

**Comment by vertxxyz** [comment]
You can implement [`IHasCustomMenu`](https://docs.unity3d.com/ScriptReference/IHasCustomMenu.html) to add to the context menu of an Editor Window tab.

**Comment by vertxxyz** [comment]
You can add a lock button to an EditorWindow using the following code:
```csharp
private static GUIStyle s_lockButton;
[SerializeField] private bool _isLocked;

private void ShowButton(Rect r)
{
    s_lockButton ??= "IN LockButton";
    using var cCS = new EditorGUI.ChangeCheckScope();
    
    _isLocked = GUI.Toggle(r, _isLocked, GUIContent.none, s_lockButton);
    
    if (cCS.changed)
        Debug.Log($"{nameof(_isLocked)} Changed");
}
```

**Comment by zORg-alex** [comment]
An `EditorWindow` is a `ScriptableObject`, if you want it to persist between sessions it can't be a nested class; it must be in its own file, sharing its name with the type.
